actions/drain_node_handler.go

```diff
@@ -183,58 +167,28 @@ func (h *drainNodeHandler) listNodePods(ctx context.Context, node *v1.Node) (*v1
 		}
 		pods = p
 		return nil
-	}, defaultBackoff(ctx))
-	return pods, err
-}
-
-func (h *drainNodeHandler) evictPods(ctx context.Context, log logrus.FieldLogger, pods []v1.Pod) error {
-	log.Infof("evicting %d pods", len(pods))
-
-	g, ctx := errgroup.WithContext(ctx)
-	for _, pod := range pods {
-		pod := pod
-
-		g.Go(func() error {
-			err := h.evictPod(ctx, pod)
-			if err != nil {
-				return err
-			}
-			return h.waitPodTerminated(ctx, log, pod)
-		})
+	}, defaultBackoff(ctx)); err != nil {
+		return nil, fmt.Errorf("listing node %v pods: %w", node.Name, err)
 	}
 
-	if err := g.Wait(); err != nil {
-		return fmt.Errorf("evicting pods: %w", err)
-	}
+	podsToEvict := lo.Filter(pods.Items, func(pod v1.Pod, _ int) bool {
+		return !isDaemonSetPod(&pod) && !isStaticPod(&pod)
+	})
 
-	return nil
+	return podsToEvict, nil
 }
 
-func (h *drainNodeHandler) waitPodTerminated(ctx context.Context, log logrus.FieldLogger, pod v1.Pod) error {
-	b := backoff.WithContext(backoff.NewConstantBackOff(5*time.Second), ctx) // nolint:gomnd
-
-	err := backoff.Retry(func() error {
-		p, err := h.clientset.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
-		if err != nil && apierrors.IsNotFound(err) {
-			return nil
-		}
+func (h *drainNodeHandler) waitNodePodsTerminated(ctx context.Context, node *v1.Node) error {
+	return backoff.Retry(func() error {
+		pods, err := h.listNodePodsToEvict(ctx, node)
 		if err != nil {
-			return err
+			return fmt.Errorf("waiting for node %q pods to be terminated: %w", node.Name, err)
 		}
-		// replicaSets will recreate pods with equal name and namespace, therefore we compare UIDs
-		if p.GetUID() == pod.GetUID() {
-			return errPodPresent
+		if len(pods) > 0 {
+			return fmt.Errorf("waiting for %d pods to be terminated on node %v", len(pods), node.Name)
 		}
 		return nil
-	}, b)
-	if err != nil && errors.Is(err, errPodPresent) {
-		log.Infof("timeout waiting for pod %s in namespace %s to terminate", pod.Name, pod.Namespace)
-		return nil
-	}
-	if err != nil {
-		return fmt.Errorf("waiting for pod %s in namespace %s termination: %w", pod.Name, pod.Namespace, err)
-	}
-	return nil
+	}, backoff.WithContext(backoff.NewConstantBackOff(10*time.Second), ctx))
```

Comment:
Maybe move this hardcoded duration to drainNodeConfig which contains more of these constants.